### PR TITLE
update app insight to store instrumentation_key on keyvault

### PR DESCRIPTION
**Pull request by rchessa**

### CODE REVIEW

File: azurerm_app_insights/main.tf [hunk 1]
```diff
@@ -3,6 +3,10 @@
 data "azurerm_resource_group" "rg" {
   name = "${local.azurerm_resource_group_name}"
 }
+data "azurerm_key_vault" "key_vault" {
```
Comment (LucaPrete) [inline azurerm_app_insights/main.tf]: Add a blank line between the two data blocks

File: azurerm_app_insights/main.tf [hunk 2]
```diff
@@ -13,3 +17,13 @@ resource "azurerm_application_insights" "azurerm_application_insights" {
...
 }
+
+resource "azurerm_key_vault_secret" "azurerm_application_insights" {
+  name         = "${var.azurerm_application_insights_instrumentation_key}"
+  value        = "${azurerm_application_insights.azurerm_application_insights.instrumentation_key}"
+  key_vault_id = "${data.azurerm_key_vault.key_vault.id}"
+
+  lifecycle {
+    ignore_changes = ["value"]
+  }
```
Comment (LucaPrete) [inline azurerm_app_insights/main.tf]: Why do we have this ignore_changes? We don't use it anywhere else when we create secrets..

File: azurerm_app_insights/variables.tf [hunk 3]
```diff
@@ -15,7 +15,14 @@ variable "applications_insights" {
   description = "A list of map(s) that contains one or application insights definition. Please read README.md to the list of keys required to be present"
 }
 
+variable "azurerm_application_insights_instrumentation_key" {
+  description = "The secret name used for app insights instrumentation key."
+  default = "fn2CommonAppInsightsInstrumentationKey"
```
Comment (LucaPrete) [inline azurerm_app_insights/variables.tf]: align =

File: azurerm_app_insights/main.tf (hunk 1 above)
Comment (rchessa) [reply azurerm_app_insights/main.tf]: done

File: azurerm_app_insights/main.tf (hunk 2 above)
Comment (rchessa) [reply azurerm_app_insights/main.tf]: ignore_changes removed

**Pull request merged**